modules/core/04-channel/keeper/upgrade.go

```diff
@@ -39,35 +38,28 @@ func (k Keeper) ChanUpgradeInit(
 		return types.Upgrade{}, err
 	}
 
-	proposedUpgrade, err := k.constructProposedUpgrade(ctx, portID, channelID, upgradeFields, upgradeTimeout)
-	if err != nil {
-		return types.Upgrade{}, errorsmod.Wrap(err, "failed to construct proposed upgrade")
-	}
-
-	channel.UpgradeSequence++
-	k.SetChannel(ctx, portID, channelID, channel)
```

Comment:
If we call `ChanUpgradeInit` as well from `ChanUpgradeTry` (to execute the logic of the function `initUpgradeHandshake` of the spec), should we keep this still here instead of adding it to `WriteUpgradeInitChannel`?

Reply:
I don't think it makes a difference because when `ChanUpgradeInit` is called from the try handler it also calls the write fn directly afterwards